File: src/SD.Infrastructure/MemberShip/LoginMenuInfo.cs
using SD.Infrastructure.Constants;
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;

namespace SD.Infrastructure.MemberShip
{
    /// <summary>
    /// 登录菜单信息
    /// </summary>
    [DataContract]
    [Serializable]
    public class LoginMenuInfo
    {
        /// <summary>
        /// 无参构造器
        /// </summary>
        public LoginMenuInfo()
        {
            this.SubMenuInfos = new HashSet<LoginMenuInfo>();
        }


        /// <summary>
        /// 信息系统编号
        /// </summary>
        [DataMember]
        public string SystemNo { get; set; }

        /// <summary>
        /// 应用程序类型
        /// </summary>
        [DataMember]
        public ApplicationType ApplicationType { get; set; }

        /// <summary>
        /// 菜单Id
        /// </summary>
        [DataMember]
        public Guid Id { get; set; }

        /// <summary>
        /// 上级菜单Id
        /// </summary>
        [DataMember]
        public Guid? ParentId { get; set; }

        /// <summary>
        /// 菜单名称
        /// </summary>
        [DataMember]
        public string Name { get; set; }

        /// <summary>
        /// 菜单排序
        /// </summary>
        [DataMember]
        public int Sort { get; set; }

        /// <summary>
        /// 菜单图标
        /// </summary>
        [DataMember]
        public string Icon { get; set; }

        /// <summary>
        /// 链接
        /// </summary>
        [DataMember]
        public string Url { get; set; }

        /// <summary>
        /// 路径
        /// </summary>
        [DataMember]
        public string Path { get; set; }

        /// <summary>
        /// 子菜单集
        /// </summary>
        [DataMember]
        public ICollection<LoginMenuInfo> SubMenuInfos { get; set; }
    }
}

File: angular/projects/sd-infrastructure/src/values/structs/login-menu-info.ts
import {ApplicationType} from "../enums/application-type";

/*登录菜单信息*/
export interface LoginMenuInfo {

    /*信息系统编号*/
    systemNo: string;

    /*应用程序类型*/
    applicationType: ApplicationType;

    /*菜单Id*/
    id: string;

    /*上级菜单Id*/
    parentId: string | null;

    /*菜单名称*/
    name: string;

    /*菜单层次*/
    level: number;

    /*菜单图标*/
    icon: string;

    /*链接*/
    url: string;

    /*路径*/
    path: string;

    /*菜单排序*/
    sort: number;

    /*是否是叶子级节点*/
    isLeaf: boolean;

    /*下级菜单列表*/
    subMenuInfos: Array<LoginMenuInfo>;
}
